src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -294,7 +295,7 @@ undistribute_table(PG_FUNCTION_ARGS)
 	EnsureCoordinator();
 	EnsureRelationExists(relationId);
 	EnsureTableOwner(relationId);
-
+	EnsureTableNotForeign(relationId);
```

Comment:
This looks fine but maybe we can add this check in the `UndistributeTable` function instead with the other checks.
I think the checks here are to see if we can use `undistribute_table` UDF but the ones in `UndistributeTable` are for checking if we can actually undistribute the table.